### PR TITLE
feat: sqlite JSON support

### PR DESCRIPTION
**Pull request by jacek-prisma**
Requires prisma/prisma#25871

Closes prisma/prisma#3786
Closes [ORM-446](https://linear.app/prisma-company/issue/ORM-446/support-for-json-field-type-in-sqlite)

### CODE REVIEW

File: quaint/src/visitor/sqlite.rs [hunk 1]
```diff
@@ -282,8 +282,21 @@ impl<'a> Visitor<'a> for Sqlite<'a> {
...
+    #[cfg(any(feature = "postgresql", feature = "mysql", feature = "sqlite"))]
+    fn visit_json_extract(&mut self, json_extract: JsonExtract<'a>) -> visitor::Result {
+        self.visit_expression(*json_extract.column)?;
+
+        if json_extract.extract_as_string {
+            self.write("->>")?;
+        } else {
+            self.write("->")?;
+        }
+        match json_extract.path.clone() {
```
Comment (jkomyno) [inline quaint/src/visitor/sqlite.rs:294]:
```suggestion
        }
  
        match json_extract.path.clone() {
```

Besides, is the clone necessary?

File: psl/psl-core/src/datamodel_connector.rs [hunk 2]
```diff
@@ -302,6 +302,19 @@ pub enum Flavour {
     Sqlite,
 }
 
+impl Flavour {
+    pub fn is_sql(&self) -> bool {
+        matches!(
+            self,
+            Flavour::Cockroach | Flavour::Sqlserver | Flavour::Mysql | Flavour::Postgres | Flavour::Sqlite
+        )
+    }
```
[the remark this answers is not in the document]
Comment (jacek-prisma) [reply psl/psl-core/src/datamodel_connector.rs]: done

File: psl/psl/tests/capabilities/sqlite.rs [hunk 3]
```diff
@@ -108,35 +108,6 @@ fn unique_index_names_support() {
...
-          [1;94m-->[0m  [4mschema.prisma:8[0m
-        [1;94m   | [0m
-        [1;94m 7 | [0m  id   Int @id
-        [1;94m 8 | [0m  [1;91mdata Json[0m
-        [1;94m 9 | [0m}
-        [1;94m   | [0m
-    "#]];
-
-    expectation.assert_eq(&error);
-}
```
Comment (jkomyno) [inline psl/psl/tests/capabilities/sqlite.rs:138]: Why did you delete this? Just adapt the expectations instead

Comment (jacek-prisma) [reply psl/psl/tests/capabilities/sqlite.rs:138]: all expectations in the file assert for errors, but I can add it back asserting for no error

File: quaint/src/visitor/sqlite.rs [hunk 4]
```diff
@@ -292,11 +305,47 @@ impl<'a> Visitor<'a> for Sqlite<'a> {
...
+                self.visit_expression(Expression::from(Value::text("true")))?;
+                self.write(" OR JSON_TYPE")?;
+                self.surround_with("(", ")", |s| s.visit_expression(left))?;
+                self.write(" = ")?;
+                self.visit_expression(Expression::from(Value::text("false")))?;
+            }
+            JsonType::Number => {
+                self.visit_expression(Expression::from(Value::text("integer")))?;
+                self.write(" OR JSON_TYPE")?;
+                self.surround_with("(", ")", |s| s.visit_expression(left.clone()))?;
```
Comment (jkomyno) [inline quaint/src/visitor/sqlite.rs:336]:
```suggestion
                self.surround_with("(", ")", |s| s.visit_expression(left))?;
```

File: query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/data_types/json.rs [hunk 5]
```diff
@@ -55,7 +55,7 @@ mod json {
 
                 insta::assert_snapshot!(
                   res,
-                  @r###"{"data":{"findManyTestModel":[{"json":"{}"},{"json":"{\"a\":\"b\"}"},{"json":"1"},{"json":"\"hello\""},{"json":"[1,\"a\",{\"b\":true}]"}]}}"###
+                  @r###"{"data":{"findManyTestModel":[{"json":"{}"},{"json":"{\"a\":\"b\"}"},{"json":"1"},{"json":"1.5"},{"json":"\"hello\""},{"json":"[1,\"a\",{\"b\":true}]"}]}}"###
```
Comment (jkomyno) [inline query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/data_types/json.rs:58]: These snapshots shouldn't have changed. What caused this?

Comment (jacek-prisma) [reply query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/data_types/json.rs:58]: I've covered an additional case in the test, returning a float directly in a JSON field, it covers a new path in the SQLite implementation because it's returned as REAL instead of JSON there

File: quaint/src/visitor/sqlite.rs (hunk 1 above)
Comment (jacek-prisma) [reply quaint/src/visitor/sqlite.rs]: removed

**Review by jkomyno — APPROVED** [review]
